### PR TITLE
Fix SessionController.SendMessageCommand Body Binding

### PR DESCRIPTION
**Pull request by BrianCArnold**
Changed SessionController.SendMessageCommand to bind information to composite object instead query string parameters.

**Changes**
Changes selected parameter binding of SendMessageCommand to a single composite object bound by [FromBody].

**Issues**
Resolves: #5628

### CODE REVIEW

File: Jellyfin.Api/Controllers/SessionController.cs [hunk 1]
```diff
@@ -301,28 +301,35 @@ public class SessionController : BaseJellyfinApiController
...
-            [FromQuery] long? timeoutMs)
+            [FromBody] MessageCommand command)
         {
-            var command = new MessageCommand
+            if (command == null)
+            {
+                throw new ArgumentException("Request body may not be null");
+            }
+
+            // Need to check if message.Text is null, since [Required] can't be applied to properties of a deserialized object.
```
Comment (cvium) [inline Jellyfin.Api/Controllers/SessionController.cs:319]: That may have been the case when json.net was the default JSON serializer, but it isn't anymore. As long as the type is nullable, [Required] will work just fine.

Comment (BrianCArnold) [reply Jellyfin.Api/Controllers/SessionController.cs:327]: I have removed the check for `command == null` and added the `[Required]` attribute to the property.

I'm currently testing to confirm that the `[Required(AllowEmptyStrings = false)]` attribute may be applied to the model property to check for null/missing/empty `Text` properties, and will update this pull request with confirmed results.

Comment (BrianCArnold) [reply Jellyfin.Api/Controllers/SessionController.cs]: As recommended, the [Required] attribute on the Text property of MessageCommand serves the necessary purpose.

File: Jellyfin.Api/Controllers/SessionController.cs [hunk 2]
```diff
@@ -313,28 +313,24 @@ public class SessionController : BaseJellyfinApiController
...
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         public ActionResult SendMessageCommand(
             [FromRoute, Required] string sessionId,
-            [FromQuery, Required] string text,
-            [FromQuery] string? header,
-            [FromQuery] long? timeoutMs)
+            [FromBody, Required] MessageCommand command)
         {
-            var command = new MessageCommand
+            var nullCorrectedCommand = new MessageCommand
```
Comment (cvium) [inline Jellyfin.Api/Controllers/SessionController.cs:326]: Is there any reason to create a new instance?

Comment (BrianCArnold) [reply Jellyfin.Api/Controllers/SessionController.cs:326]: It's a force of habit not to modify a reference type parameter's properties. Doesn't apply here, I'll modify.

Comment (cvium) [reply Jellyfin.Api/Controllers/SessionController.cs]: Normally you shouldn't alter the input but the object is a result of model binding, so it's fine imo

File: Jellyfin.Api/Controllers/SessionController.cs [hunk 3]
```diff
@@ -313,28 +313,22 @@ public class SessionController : BaseJellyfinApiController
...
+            [FromBody, Required] MessageCommand command)
         {
-            var command = new MessageCommand
+            if (string.IsNullOrWhiteSpace(command.Header)) 
             {
-                Header = string.IsNullOrEmpty(header) ? "Message from Server" : header,
-                TimeoutMs = timeoutMs,
-                Text = text
-            };
+                command.Header =  "Message from Server";
```
Comment (cvium) [inline Jellyfin.Api/Controllers/SessionController.cs:328]:
```suggestion
                command.Header = "Message from Server";
```

File: Jellyfin.Api/Controllers/SessionController.cs [hunk 4]
```diff
@@ -313,28 +313,22 @@ public class SessionController : BaseJellyfinApiController
...
+            }
 
-            _sessionManager.SendMessageCommand(RequestHelpers.GetSession(_sessionManager, _authContext, Request).Id, sessionId, command, CancellationToken.None);
+            _sessionManager.SendMessageCommand(RequestHelpers.GetSession(_sessionManager, _authContext, Request).Id, sessionId, nullCorrectedCommand, CancellationToken.None);
```
Comment (cvium) [inline Jellyfin.Api/Controllers/SessionController.cs:331]:
```suggestion
            _sessionManager.SendMessageCommand(RequestHelpers.GetSession(_sessionManager, _authContext, Request).Id, sessionId, command, CancellationToken.None);
```

**Comment by BrianCArnold** [comment]
Thanks, I need to stop making commits before I've had my coffee.

**Review by cvium — APPROVED** [review]

**Pull request merged**